core/trino-main/src/main/java/io/trino/eventlistener/EventListenerManager.java

```diff
@@ -218,4 +219,17 @@ public TimeStat getSplitCompletedTime()
     {
         return splitCompletedTime;
     }
+
+    @PreDestroy
+    public void shutdown()
+    {
+        for (EventListener listener : configuredEventListeners.get()) {
```

Comment:
If we are shutting down event listener manager than I think we should actually make it deconfigure existing listeners.

This line should be:

```
for (EventListener listener : configuredEventListeners.getAndSet(List.of())) {
```

which returns previously configured event listeners and swaps them with empty list.

@electrum @findepi @kokosing wdyt?